upstairs/src/lib.rs

```diff
@@ -2905,7 +3341,27 @@ impl Upstairs {
         client_id: u8,
         read_data: Result<Vec<ReadResponse>, CrucibleError>,
     ) -> Result<bool> {
+        /*
+         * We can't get the upstairs state lock when holding the downstairs
+         * lock, but we need to make decisions about this downstairs work
+         * knowing the upstairs state.  So,
+         *  * get the upstairs state lock,
```

Comment:
grabbing multiple locks in a certain order :grimacing: is there something else we can do here? or am I being too paranoid?

Reply:
I wish I knew a better way to enforce this, and I'm open to suggestions.  It makes me nervous as well as I created several deadlocks while developing this.  The current best I could think of was this comment :)